editor/src/components/canvas/dom-walker.ts

```diff
@@ -1081,15 +1076,25 @@ function getClosestOffsetParent(element: HTMLElement): Element | null {
   return null
 }
 
-function measureGlobalFramesOfGridCellsFromControl(
-  gridPath: ElementPath,
+function measureGlobalFramesOfGridCells(
+  element: HTMLElement,
   scale: number,
   containerRectLazy: CanvasPoint | (() => CanvasPoint),
   elementCanvasRectangleCache: ElementCanvasRectangleCache,
 ): GridCellGlobalFrames | null {
-  const path = EP.toString(gridPath)
+  const paths = getPathsOnDomElement(element)
+
   let gridCellGlobalFrames: GridCellGlobalFrames | null = null
-  const gridControlElement = document.getElementById(GridMeasurementHelperKey(EP.fromString(path)))
+  const gridControlElement = (() => {
+    for (let p of paths) {
```

Comment:
Does this work with nested grids?

Reply:
Absolutely! These paths are all utopia element paths which render exactly to this html element, so there is no overlap between the paths of different grids.